sensing/src/main/java/com/google/android/sensing/upload/SensingSynchronizer.kt

```diff
@@ -40,77 +44,96 @@ class SensingSynchronizer(
    * [issue](https://github.com/google/android-fhir/issues/2119).
    */
   suspend fun synchronize(): Flow<SyncUploadState> = flow {
-    val uploadRequestList =
-      (sensingEngine.listUploadRequest(status = RequestStatus.UPLOADING) +
-        sensingEngine.listUploadRequest(status = RequestStatus.PENDING))
+    var uploadRequestList = uploadRequestFetcher.fetch()
+    emit(SyncUploadState.Started(uploadRequestList.size))
+
+    var completedSyncRequests = 0
+    var totalSyncRequests = uploadRequestList.size
+
+    while (uploadRequestList.isNotEmpty()) {
+      var batchSyncUploadState =
+        SyncUploadState.BatchState.Started(totalBatchRequests = uploadRequestList.size)
+          as SyncUploadState.BatchState
+      emit(batchSyncUploadState)
 
-    var syncUploadState = SyncUploadState.Started(uploadRequestList.size) as SyncUploadState
-    emit(syncUploadState)
-    if (uploadRequestList.isNotEmpty()) {
       // Following line is a workaround to calculate new states based on previous "InProgress"
       // states.
-      syncUploadState =
-        SyncUploadState.InProgress(
-          totalRequests = uploadRequestList.size,
+      batchSyncUploadState =
+        SyncUploadState.BatchState.InProgress(
```

Comment:
I am assuming this has added some more complexity by bringing a nested state here `SyncUploadState.BatchState.InProgress`. Lets connect to discuss on this.

Reply:
Reverted back. 
Just refactored variables names for example `currentTotalRequests` which suggests that it can change.